envoy/http/filter.h

```diff
@@ -302,6 +302,21 @@ class StreamFilterCallbacks {
    * Called when filter activity indicates that the stream idle timeout should be reset.
    */
   virtual void resetIdleTimer() PURE;
+
+  /**
+   * This is a helper to get the route's per-filter config if it exists, otherwise the virtual
+   * host's. Or nullptr if none of them exist.
+   */
+  virtual const Router::RouteSpecificFilterConfig* mostSpecificPerFilterConfig() const PURE;
```

Comment:
Isn't there already API for that https://github.com/envoyproxy/envoy/blob/c92c0326628fc75a6bcc15ae8e0d5448ca2a9c45/source/common/http/utility.h#L543 ?

Reply:
Yeah. But it just is a help function that used to get per filter config from route and is keeped for some historical reson.

The new APIs will be used as a proxy to implement the fallback config searching policy. Finally, all the filters need to use these new APIs.